packages/gateway/src/gateway-language-model.ts

```diff
@@ -71,7 +71,11 @@ export class GatewayLanguageModel implements LanguageModelV2 {
         warnings: [],
       };
     } catch (error) {
-      throw asGatewayError(error);
+      const headers = await resolve(this.config.headers());
+      const authMethod = headers['x-ai-gateway-auth-method'] as
+        | 'api-key'
+        | 'oidc';
+      throw asGatewayError(error, authMethod);
```

Comment:
@lgrammel guidance here (more in PR, I will comment) is generally to inline as much as possible for better readability/maintenance of code, so:

```
throw asGatewayError(error, authMethod: (await resolve(this.config.headers()))headers['x-ai-gateway-auth-method']
```

also `as ...` preferred to avoid, it is a forced cast basically, maybe we zod-schema-parse the value to what we know is permitted, you can find examples of zod use for similar in the code base